cmd/swarm/clientaccountmanager.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+/*
+//This module implements the bare minimum to allow client commands to access the keystore
+*/
+package main
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/accounts/usbwallet"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	cli "gopkg.in/urfave/cli.v1"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+)
+
+type ClientAccountManagerProvider struct {
+	manager *accounts.Manager
+}
+
+func (c *ClientAccountManagerProvider) AccountManager() *accounts.Manager {
+	return c.manager
+}
+
+func getClientAccount(ctx *cli.Context) *ecdsa.PrivateKey {
+	bzzconfig, err := buildConfig(ctx)
+	if err != nil {
+		utils.Fatalf("Cannot read configuration: %s", err.Error())
+		return nil
+	}
+	nodeconfig := defaultNodeConfig
+	if _, err := os.Stat(bzzconfig.Path); err == nil {
+		nodeconfig.DataDir = bzzconfig.Path
+	}
+
+	accountManager, _, err := makeAccountManager(&nodeconfig)
+	if err != nil {
+		utils.Fatalf("Cannot create account manager: %s", err.Error())
+		return nil
+	}
+
+	return getAccount(bzzconfig.BzzAccount, ctx, &ClientAccountManagerProvider{
+		manager: accountManager,
+	})
+
+}
+
+func makeAccountManager(conf *node.Config) (*accounts.Manager, string, error) {
```

Comment:
This needs to be refactored, maybe a different PR. The reason this is here and with this name is because it is a blatant copy of `node.makeAccountManager`, a private method of the node package and I didn't want to touch that package. Our problem is that the resource cli is the only command that uses accounts and private keys, and all the code to handle that is part of the Swarm node stuff.

A quick solution if you think others would agree is simply making that method public (?)

We would need a refactor whereupon account management is extracted out of the node domain and can be shared by the cli as well.

What do you think?

Reply:
> We would need a refactor whereupon account management is extracted out of the node domain and can be shared by the cli as well.

Can we theoretically just sign with the RPC call?

Reply:
You need a private key. The idea of copying this was to reuse the account management code. So people using the `swarm` binary as a client can benefit of having their private keys in the same way as if they were running a node.

We can also disable this, but then users would need to point `--bzzaccount` to an unencrypted private key file, or we could have an additional parameter where the user can specify the path to their private key.

Reply:
This sounds like a wider discussion. Let's have a phone about this when all of this is done.

Reply:
Yes, I agree